### PR TITLE
Fix setting world size and rank in distributed samplers

### PR DESCRIPTION
**Pull request by pzelasko**
Fixes issue introduced in #213

### CODE REVIEW

**Review by danpovey — COMMENTED** [review]
LGTM!

**Review by danpovey — APPROVED** [review]
forgot to approve

File: lhotse/dataset/sampling.py [hunk 1]
```diff
@@ -97,12 +97,8 @@ def _maybe_init_distributed(self, world_size: Optional[int], rank: Optional[int]
             self.world_size = 1
             self.rank = 0
             return
-
-        if world_size is None:
-            self.world_size = dist.get_world_size()
-
-        if rank is None:
-            self.rank = dist.get_rank()
+        self.world_size = dist.get_world_size() if world_size is None else world_size
```
Comment (csukuangfj) [inline lhotse/dataset/sampling.py:100]: I find that some people is using
```python
self.world_size = world_size or dist.get_world_size()
self.rank = rank or dist.get_rank()
```
which is shorter I think, though the current approach is also OK.

**Comment by danpovey** [comment]
That evaluates it in a Boolean context, which I think is less safe:

```

>>> a = False

>>> b = a or 2

>>> b

2

>>>

```

File: lhotse/dataset/sampling.py (hunk 1 above)
Comment (pzelasko) [reply lhotse/dataset/sampling.py:100]: it looks fancy, but in this context it has a potential for a bug when `rank` is set to 0 but `dist.get_rank()` returns something else

**Pull request merged**